File: backend/Ledgerly.Application/DTOs/Payment/CreatePaymentRequest.cs
namespace Ledgerly.Application.DTOs.Payment;

public class CreatePaymentRequest
{
    public Guid BillId { get; set; }
    public Guid CustomerId { get; set; }
    public decimal Amount { get; set; }
    public DateTime PaymentDate { get; set; }
    public string? Reference { get; set; }
    public Guid? ReceiptFileId { get; set; }
}

File: app/lib/models/payment/create_payment_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'create_payment_request.g.dart';

@JsonSerializable()
class CreatePaymentRequest {
  final String billId;
  final String customerId;
  final double amount;
  @JsonKey(name: 'paymentDate')
  final DateTime paymentDate;
  final String? reference;
  final String? receiptFileId;

  CreatePaymentRequest({
    required this.billId,
    required this.customerId,
    required this.amount,
    required this.paymentDate,
    this.reference,
    this.receiptFileId,
  });

  factory CreatePaymentRequest.fromJson(Map<String, dynamic> json) =>
      _$CreatePaymentRequestFromJson(json);

  Map<String, dynamic> toJson() => _$CreatePaymentRequestToJson(this);
}

